src/data/pnat/find.lean

```diff
@@ -0,0 +1,133 @@
+/-
+Copyright (c) 2022 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky, Floris van Doorn
+-/
+import data.pnat.basic
+
+/-!
+# Explicit least witnesses to existentials on positive natural numbers
+
+Implemented via calling out to `nat.find`.
+
+-/
+
+namespace pnat
+
+variables {p q : ℕ+ → Prop} [decidable_pred p] [decidable_pred q] (h : ∃ n, p n)
+
+instance decidable_pred_exists_nat :
+  decidable_pred (λ n' : ℕ, ∃ (n : ℕ+) (hn : n' = n), p n) := λ n',
+decidable_of_iff' (∃ (h : 0 < n'), p ⟨n', h⟩) $ subtype.exists.trans $
+  by simp_rw [subtype.coe_mk, @exists_comm (_ < _) (_ = _), exists_prop, exists_eq_left']
+
+
+include h
+
+/--
```

Comment:
This docstring is attached to `find_aux`. Did you mean for it to be a file-level doc?
```suggestion
/-!
```

Reply:
I did not mean for it to be file-level doc. I don't want people to use `pnat.find_aux` and so the docstring just mentions the true API functions.